framework/src/main/java/org/tron/core/net/message/handshake/HelloMessage.java

```diff
@@ -127,17 +129,23 @@ public String toString() {
             .append("lowestBlockNum: ").append(helloMessage.getLowestBlockNum()).append("\n");
 
     ByteString address = helloMessage.getAddress();
-    if (address != null && !address.isEmpty()) {
+    if (!address.isEmpty()) {
       builder.append("address:")
               .append(StringUtil.encode58Check(address.toByteArray())).append("\n");
     }
 
     ByteString signature = helloMessage.getSignature();
-    if (signature != null && !signature.isEmpty()) {
+    if (!signature.isEmpty()) {
       builder.append("signature:")
               .append(signature.toByteArray().length).append("\n");
     }
 
+    ByteString codeVersion = helloMessage.getCodeVersion();
```

Comment:
Will this codeVersion be null when received a HelloMessage from a node with previous version ?

Reply:
Every column in proto has default value. `codeVersion` has default value `byte[0]`, not null.